CavalcanteLucas/labdale
# Feature/OL42-todo_list_delete

**Link to task:** 
https://labcodes.atlassian.net/browse/OL-42

**How to test:**
- With a user logged in having one or more todo lists, click on the "delete" button besides the todolist title, check if modal for confirmation pops up;
- Confirm delete on the open modal and check if todo list is deleted.

**Description of your solution:**
This PR implements the API endpoint for enabling DELETE requests of a Todo List instance. This PR also creates the interfacte to action that fetches the DELETE endpoint and handle state updates / redirect actions for the REQUEST/SUCCESS/FAILURE cases. Tests ensure that deleting todo lists requires Token authorization, and that it is a private method among different users. A bin icon besides the title of a todo list on the Dashboard Page represents the delete button. When this button is selected, a modal for confirmation is open. After confirmation, that instance of todo list is deleted and user is redirected to dashboard page.
